<?xml version="1.0"?>
<!-- Provided by sleutene for convenience -->
<launch>
  <!-- Launch simulator. -->
  <!-- We resume the logic in empty_world.launch, changing only the name of the world to be launched -->
  <!--<include file="$(find sjtu_drone)/launch/simple.launch">
    <arg name="world_name" value="$(find sjtu_drone)/worlds/skokloster_castle.world"/>
  </include>-->
  
  <!-- Emulate controller parameters-->
  <group ns="ardrone_driver">
    <param name="euler_angle_max" value="0.21" />
    <param name="control_vz_max" value="700" />
    <param name="control_yaw" value="1.75" />
  </group>
  
  <!-- Launch visualisation. -->
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find rmil_drone_practicals)/config/rviz_control.rviz"/>
  <!-- Launch estimator / control arp_node. -->
  <node name="rmil_drone_node" pkg="rmil_drone_practicals" type="rmil_drone_node"
      output="screen">
    <!--<remap from="ardrone/front/image_raw" to="drone/front_camera/image_raw"/>
    <remap from="ardrone/imu" to="drone/imu"/> -->
    <remap from="ardrone/front/image_raw" to="ardrone/image_raw"/>
    <remap from="ardrone/takeoff" to="drone/takeoff"/> 
    <remap from="ardrone/reset" to="drone/reset"/>
    <remap from="ardrone/land" to="drone/land"/> 
    <remap from="ardrone/navdata" to="drone/navdata"/> 
    <!-- Camera parameters-->
    <param name="fu" value="582.13726" />
    <param name="fv" value="579.40778" />
    <param name="cu" value="315.61947" />
    <param name="cv" value="205.41823" />
    <param name="k1" value="-0.563809" />
    <param name="k2" value="0.362871" />
    <param name="p1" value="-0.011535" />
    <param name="p2" value="-0.001087" />
    <param name="focalLengthMap" value="390.598" />
    <rosparam param="T_SC">[-0.00195087, -0.03257782,  0.99946730, 0.17409445, -0.99962338, -0.02729525, -0.00284087, 0.02255834, 0.02737326, -0.99909642, -0.03251230, 0.00174723, 0.00000000,  0.00000000,  0.00000000, 1.00000000]</rosparam>
    <!-- Make sure to match the map to the world loaded above! -->
    <param name="map" value="rmil_arena_map_06_06_23.csv" />
    <param name="briskMatchingThreshold" value="60" />
    <param name="briskUniformityRadius" value="25.0" />
    <param name="briskOctaves" value="0" />
    <param name="briskAbsoluteThreshold" value="50.0" />
    <param name="briskMaxNumKeypoints" value="1000" />
    <param name="reprojectionDistanceThreshold" value="5.0" />

  </node>

  <node pkg="rosbag" type="play" name="player" output="screen" args=" --clock -r 1.0 /home/barbas/Downloads/rmil_test_bag.bag"/>

</launch>
